src/main/java/jenkins/branch/OrganizationFolder.java

```diff
@@ -177,12 +185,16 @@ public OrganizationFolder(ItemGroup parent, String name) {
     @Override
     public void onCreatedFromScratch() {
         super.onCreatedFromScratch();
-        for (MultiBranchProjectFactoryDescriptor d : ExtensionList.lookup(MultiBranchProjectFactoryDescriptor.class)) {
-            MultiBranchProjectFactory f = d.newInstance();
-            if (f != null) {
-                projectFactories.add(f);
+
+        if( projectFactories.isEmpty() ) {
+            for (MultiBranchProjectFactoryDescriptor d : ExtensionList.lookup(MultiBranchProjectFactoryDescriptor.class)) {
```

Comment:
Changed Behavior:

Only add default project factories if there aren't any. This allows code that creates an OrganizationFolder to have confidence that if it sets projectFactories on the folder, the folder will use only and exactly those factories.

Without this change, default factories are always _added_ to whichever factories were set by code that created the folder. This can lead to undesired jobs being created.